set interface variable from function output parameter

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

type T struct{}

func newT() (T, error) { return T{}, nil }

func main() {
    var (
        i   interface{}
        err error
    )
    i, err = newT()
    fmt.Println(i, err)
}
```

Expected result:
```console
$ go run ./sample.go
{} <nil>
```

Got:
```console
$ yaegi ./sample.go
/home/marc/tmp/sample.go:7:33: panic
/home/marc/tmp/sample.go:11:3: panic
reflect.Set: value of type struct {} is not assignable to type interp.valueInterface
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0003ae000, 0x40)
        /home/marc/sdk/go1.14.1/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc0000a5e10)
        /home/marc/dev/yaegi/interp/interp.go:320 +0xbc
panic(0xc94360, 0xc000114910)
        /home/marc/sdk/go1.14.1/src/runtime/panic.go:967 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0003860a0, 0xc00039c100)
        /home/marc/dev/yaegi/interp/run.go:113 +0x20c
panic(0xc94360, 0xc000114910)
        /home/marc/sdk/go1.14.1/src/runtime/panic.go:967 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc000386140, 0xc000399600)
        /home/marc/dev/yaegi/interp/run.go:113 +0x20c
panic(0xc94360, 0xc000114910)
        /home/marc/sdk/go1.14.1/src/runtime/panic.go:967 +0x166
reflect.Value.assignTo(0xcc3de0, 0x158d8c8, 0x199, 0xdc618a, 0xb, 0xd17a20, 0x0, 0xc000399600, 0xcc3de0, 0x158d8c8)
        /home/marc/sdk/go1.14.1/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0xd17a20, 0xc000118a60, 0x199, 0xcc3de0, 0x158d8c8, 0x199)
        /home/marc/sdk/go1.14.1/src/reflect/value.go:1532 +0xbd
github.com/containous/yaegi/interp._return.func4(0xc000386140, 0xc0001189c0)
        /home/marc/dev/yaegi/interp/run.go:1573 +0x97
github.com/containous/yaegi/interp.runCfg(0xc000399600, 0xc000386140)
        /home/marc/dev/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.call.func5(0xc0003860a0, 0xc00037f500)
        /home/marc/dev/yaegi/interp/run.go:807 +0xac9
github.com/containous/yaegi/interp.runCfg(0xc00039c100, 0xc0003860a0)
        /home/marc/dev/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc000102600, 0xc000399900, 0xc000386000)
        /home/marc/dev/yaegi/interp/run.go:97 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000102600, 0xc000390000, 0xb7, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/marc/dev/yaegi/interp/interp.go:388 +0x41f
main.main()
        /home/marc/dev/yaegi/cmd/yaegi/yaegi.go:94 +0x548
```

--- a/_test/fun13.go
+++ b/_test/fun13.go
@@ -1,0 +1,19 @@
+package main
+
+import "fmt"
+
+type T struct{}
+
+func newT() (T, error) { return T{}, nil }
+
+func main() {
+	var (
+		i   interface{}
+		err error
+	)
+	i, err = newT()
+	fmt.Println(i, err)
+}
+
+// Output:
+// {} <nil>

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -542,22 +542,22 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 		case assignXStmt:
 			wireChild(n)
 			l := len(n.child) - 1
-			switch n.child[l].kind {
+			switch lc := n.child[l]; lc.kind {
 			case callExpr:
 				n.gen = nop
 			case indexExpr:
-				n.child[l].gen = getIndexMap2
+				lc.gen = getIndexMap2
 				n.gen = nop
 			case typeAssertExpr:
 				if n.child[0].ident == "_" {
-					n.child[l].gen = typeAssertStatus
+					lc.gen = typeAssertStatus
 				} else {
-					n.child[l].gen = typeAssert2
+					lc.gen = typeAssert2
 				}
 				n.gen = nop
 			case unaryExpr:
-				if n.child[l].action == aRecv {
-					n.child[l].gen = recv2
+				if lc.action == aRecv {
+					lc.gen = recv2
 					n.gen = nop
 				}
 			}

--- a/interp/run.go
+++ b/interp/run.go
@@ -670,7 +670,12 @@ func call(n *node) {
 	case defineXStmt, assignXStmt:
 		for i := range rvalues {
 			c := n.anc.child[i]
-			if c.ident != "_" {
+			switch {
+			case c.ident == "_":
+				// Skip assigning return value to blank var.
+			case c.typ.cat == interfaceT:
+				rvalues[i] = genValueInterfaceValue(c)
+			default:
 				rvalues[i] = genValue(c)
 			}
 		}

